### ISSUE TITLE
Remove python 2.6 warning from FilePath.__init__

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8162]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#8162|
|Type|enhancement|
|Created|2015-12-31 14:22:23Z|
|Branch|https://github.com/twisted/twisted/tree/filepath-py2.6-clean-8162|

since Python 2.6 is no longer supported this can be removed.

I need it so that I can use FilePath together with future.types.newbytes

<details><summary>Searchable metadata</summary>

```
trac-id__8162 8162
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__None None
branch__branches_filepath_py2_6_clean_8162 branches/filepath-py2.6-clean-8162
branch_author__adiroiban adiroiban
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1451571743107990 1451571743107990
changetime__1451655959468321 1451655959468321
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

(In [46579]) Branching to filepath-py2.6-clean-8162.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban removed owner|
|-|-|

I have removed the python 2.6 handling code and its tests.

I put it under .misc since 2.6 is no longer support so I don't see any point in having the NEWS file filled with notes about Python 2.6 cleanup.

Please check the changes.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml set owner to @adiroiban|
|-|-|

Looks good to me. Agree re misc file. OK to merge.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set status to `closed`|
|-|-|

(In [46581]) Merge filepath-py2.6-clean-8162: Remove Python 2.6 warning from FilePath.__init__

Author: adiroiban
Reviewer: jml
Fixes: [#8162](https://github.com/twisted/twisted/issues/8162)
